### PR TITLE
Branch delete reminder

### LINKED ISSUE #75: Delete reminder

### PR DESCRIPTION
**Pull request by sarahteonin**
Fixes #75

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/DeleteAppointmentCommandParser.java [hunk 1]
```diff
@@ -16,11 +21,19 @@ public class DeleteAppointmentCommandParser implements Parser<DeleteAppointmentC
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteAppointmentCommand parse(String args) throws ParseException {
-        try {
-            Name name = ParserUtil.parseName(args);
-            return new DeleteAppointmentCommand(name);
-        } catch (ParseException pe) {
-            throw new ParseException(MESSAGE_INVALID_NAME_DISPLAYED, pe);
+        requireNonNull(args);
```
Comment (GuanpengR) [inline src/main/java/seedu/address/logic/parser/DeleteAppointmentCommandParser.java:24]: Nice practice of defensive coding!

File: src/test/java/seedu/address/logic/commands/DeleteReminderCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        String expectedMessage = String.format(DeleteReminderCommand.MESSAGE_DELETE_REMINDER_SUCCESS,
+                Messages.formatReminder(personWithReminderToDelete,
+                        personWithReminderToDelete.getReminder().toString()));
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.deleteReminder(personWithReminderToDelete);
+
+        assertCommandSuccess(deleteReminderCommand, model, expectedMessage, expectedModel);
+    }
+     */
```
Comment (GuanpengR) [inline src/test/java/seedu/address/logic/commands/DeleteReminderCommandTest.java:64]: Maybe consider removing unused test cases?

File: src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java [hunk 3]
```diff
@@ -48,10 +55,31 @@ public CommandResult execute(Model model) throws CommandException {
...
 
-        Person appointmentToDelete = lastShownList.get(index);
-        model.deleteAppointment(appointmentToDelete);
+        Person personWithAppointmentToDelete = lastShownList.get(index);
+
+        Schedule appointmentToDelete = new Schedule("", "");
+        for (Schedule schedule : personWithAppointmentToDelete.getSchedules()) {
+            if (schedule.equals(appointment)) {
+                appointmentToDelete = appointment;
+                break;
```
Comment (GuanpengR) [inline src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java:64]: Nice making use of the uniqueness of appointment to optimize the for loop!

File: src/main/java/seedu/address/logic/Messages.java [hunk 4]
```diff
@@ -14,8 +17,9 @@ public class Messages {
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The name provided is invalid";
-    public static final String MESSAGE_INVALID_NAME_DISPLAYED = "The name provided is invalid";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid.";
+    public static final String MESSAGE_INVALID_NAME_DISPLAYED = "The name provided is not in the address book.";
+    public static final String MESSAGE_INVALID_APPOINTMENT_DISPLAYED = "The appointment provided is invalid.";
```
Comment (GuanpengR) [inline src/main/java/seedu/address/logic/Messages.java:22]: The error messages are more precise now. Good Job!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -139,6 +139,20 @@ public static Schedule parseSchedule(String dateTime, String note) throws ParseE
...
     }
 
+    /**
+     * Parses a {@code String date} into a {@code Schedule} with an empty note.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code Schedule} is invalid.
+     */
+    public static Schedule parseSchedule(String dateTime) throws ParseException {
+        requireNonNull(dateTime);
```
Comment (jasmiinee) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:149]: Good practice for defensive coding!

**Pull request merged**